lib/FirmwareModule.js

```diff
@@ -95,7 +95,12 @@ FirmwareModule.prototype = extend(FirmwareModule.prototype, {
 		}
 	},
 
-	areDependenciesMet: function(deviceModules) {
+	areDependenciesMet: function(deviceModules, invalidModules) {
+		invalidModules = invalidModules || [];
+
+		// Make sure the array is empty
+		invalidModules.length = 0;
```

Comment:
Even if this is a well-defined behavior of an array object (it looks very unusual for me though), I'd remove this and simply assume that the calling code either passes an empty `invalidModules` or knows what it's doing

Reply:
(I'm assuming that we cannot change the API and make this function return the array in question)